src/typechecker/FStar.TypeChecker.TcTerm.fst

```diff
@@ -2336,25 +2367,76 @@ and check_application_args env head (chead:comp) ghead args expected_topt : term
       in
 
       (* 1. We compute the final computation type comp  *)
+
+      //
```

Comment:
Make sure that the computation types are closed in the environment when we call `bind`.